app/controllers/applicationController.js

```diff
@@ -70,11 +95,43 @@ updateApplication = async (req, res) => {
                     message: "There was an error updating your application"
                 });
             })
+    } else {
+        res.status(404).json({
+            message: `Application ${id} not found`,
+        });
     }
 };
 
 deleteApplication = async (req, res) => {
+    const id = req.params.id;
 
+    const application = await Application.findById(id).catch(() => {
+        res.status(404).json({
+            message: `Application ${id} not found`,
+        });
+    });
+    if (res.statusCode == 404) {
+        return;
+    }
+    console.log(application);
+    if(application) {
+        Application.deleteOne({ _id: application._id })
+        .then(() => {
+            return res.status(200).json({
+                message: `Application deleted`,
+            });
+        })
+        .catch(error => {
+            return res.status(400).json({
+                error,
+                message: "There was an error deleting your application"
+            });
+        })
+    } else {
+        res.status(201).json({
```

Comment:
```suggestion
        res.status(404).json({
```

For consistency

Reply:
This was done intentionally since it's a delete. The 201 is thrown when it's a valid id but there is no db object. Thus it's already "deleted"

Reply:
Oh is 204 no response?

Reply:
Ah my bad should be 204 - No content

Reply:
It's the same handling we're doing in update. If the requested application isn't found on the db it should return a response code indicating that